File: submissions/week_3/6_maximum_number_of_prizes/different_summands.R
# Stanley Yu

optimal_summands <- function(n) {
  summands = c() 
  if (n > 2) {
    prize_value = 1 
    while (prize_value <= 2) {
      summands = append(summands, prize_value)
      prize_value = prize_value + 1 
    }
    summands = append(summands, n - prize_value)
  } else {
    summands = append(summands, n)
  }
  return(summands)
}


print("n = 6")
optimal_summands(6)

print("n=8")
optimal_summands(8)

print("n=2")
optimal_summands(2)

File: submissions/week_3/6_maximum_number_of_prizes/different_summands.py
# Uses python3
import sys

def optimal_summands(n):
    summands = []
    #write your code here        
    if n > 2: 
        prize_value = 1
        while prize_value <= 2: 
            summands.append(prize_value) 
            prize_value += 1
        summands.append(n - prize_value)
    else: 
        summands.append(n)
            
    return summands

if __name__ == '__main__':
    input = sys.stdin.read()
    n = int(input)
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print(x, end=' ')
